src/omotes_sdk/internal/worker/worker.py

```diff
@@ -203,13 +217,27 @@ def wrapped_worker_task(
 
     :param task: Reference to the CeleryTask.
     :param job_id: Id of the job this task belongs to.
+    :param job_reference: A user-submitted reference to this job.
     :param input_esdl: ESDL description which needs to be processed.
     :param params_dict: job, non-ESDL, parameters.
     """
     logger.info("Worker started new task %s", job_id)
     task_util = TaskUtil(job_id, task, task.broker_if)
     task_util.update_progress(0, "Job calculation started")
-    task.output_esdl = WORKER_TASK_FUNCTION(input_esdl, params_dict, task_util.update_progress)
+    output_esdl = WORKER_TASK_FUNCTION(input_esdl, params_dict, task_util.update_progress)
+
+    output_esh = pyesdl_from_string(output_esdl)
+    output_energy_system: EnergySystem = output_esh.energy_system
+    input_esh = pyesdl_from_string(input_esdl)
+    input_energy_system: EnergySystem = input_esh.energy_system
+    if job_reference is None:
+        output_energy_system.name = f"{input_energy_system.name}_{WORKER_TASK_TYPE}"
+    elif job_reference == "":
+        output_energy_system.name = f"{input_energy_system.name}"
+    else:
+        output_energy_system.name = f"{input_energy_system.name}_{job_reference}"
```

Comment:
should we replace invalid xml characters with `_` for example?

Reply:
Good catch! Checked what happened when I inserted reserved characters in the job reference. Pyesdl already takes care of the necessary escaping:
```xml
<esdl:EnergySystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:esdl="http://www.tno.nl/esdl" id="c53b2878-6f0e-49fb-b9a1-9aeb8b808117" description="" name="Untitled EnergySystem with return network_example_sdk_client.simulator %&gt;&gt;&amp;#(@#)(@#$&lt;&gt;)(ASd&lt;&lt;&lt;&gt;&gt;&gt;&gt;&gt;" version="1" esdlVersion="v2207">
```